AmplifyTools/amplify-tools.sh

```diff
@@ -6,13 +6,121 @@
 # SPDX-License-Identifier: Apache-2.0
 
 set -e
-export PATH=$PATH:$(npm bin -g)
 
 if ! which node >/dev/null; then
   echo "warning: Node is not installed. Visit https://nodejs.org/en/download/ to install it"
   exit 1
-elif ! test -f ./amplifytools.xcconfig; then
-  npx amplify-app --platform ios
+fi
+
+export PATH=$PATH:$(npm bin -g)
+
+# Note the use of tail -1 is important here because when upgrading between versions
+# the first time that you run these commands, we have seen this variable take on the value of:
+# """
+# Scanning for plugins...
+# plugin scan successful
+# 4.21.0
+# """
+AMP_APP_VERSION_CURRENT=`npx -q amplify-app --version |tail -1`
+AMP_APP_VERSION_MINIMUM="2.17.1"
+AMP_APP_VERSION_INVALID=0
+
+AMP_CLI_VERSION_CURRENT=`npx -q amplify --version |tail -1`
+AMP_CLI_VERSION_MINIMUM="4.21.0"
+AMP_CLI_VERSION_INVALID=0
+
+
+STRIP_ESCAPE_RESULT=
+stripEscapeUtil() {
+    STRIP_ESCAPE_RESULT=
+
+    set +e
+    HAS_RUBY=0
+    which ruby > /dev/null 2>&1
+    if [[ $? -eq 0 ]]; then
+	HAS_RUBY=1
+    fi
+    set -e
+
+    if [ ${HAS_RUBY} -eq 1 ]; then
+	STRIP_ESCAPE_RESULT=`echo "${1}" | ruby -pe 'gsub(/\e\[[0-9;]*m/s, "")'`
+    else
+	STRIP_ESCAPE_RESULT=`echo "${1}" | npx -q strip-ansi-cli`
+    fi
+}
+
+VERSION_INVALID=
+checkMinVersionCompatibility() {
+    VERSION_INVALID=0
+
+    stripEscapeUtil "${1}"
```

Comment:
nit: It seems reasonable to make the caller responsible for passing a well-formed argument, rather than the function having to deal with escaped input

Reply:
Going to keep this as is for this reason in example B.  That is, I want to make sure I am preserving the argument I am sending to another function.

Example A:
```
func2() {
    echo "Number of args is: $#"
}
func1() {
    func2 ${1}
}
func1 "hello wooj2"
```
Outputs: 2


Example B:
```
func2() {
    echo "Number of args is: $#"
}
func1() {
    func2 "${1}"
}
func1 "hello wooj2"
```
Outputs: 1